FEXCore/Source/Utils/ArchHelpers/Arm64.cpp

```diff
@@ -2016,6 +2016,29 @@ HandleUnalignedAccess(FEXCore::Core::InternalThreadState* Thread, UnalignedHandl
         LogMan::Msg::EFmt("Unhandled JIT SIGBUS LDLUR*: PC: 0x{:x} Instruction: 0x{:08x}\n", ProgramCounter, PC[0]);
         return std::nullopt;
       }
+    } else if (!IsJIT) {
+      if ((Instr & ArchHelpers::Arm64::LDAXR_MASK) == ArchHelpers::Arm64::LDAXR_INST) { // LDAXR*
+        if (ArchHelpers::Arm64::HandleAtomicLoad(Instr, GPRs, 0, &Thread->ExclusiveStore)) {
+          return 4;
+        }
+      } else if ((Instr & ArchHelpers::Arm64::STLXR_MASK) == ArchHelpers::Arm64::STLXR_INST) { // STLXR*
+        uint32_t StatusReg = Instr << 11 >> 27;
+        // // Emulate exclusive store by validating the address and value against the last unaligned LDAXR*.
+        if (GPRs[AddrReg] != Thread->ExclusiveStore.Addr || Size > Thread->ExclusiveStore.Size) {
```

Comment:
I don't think `Size > Thread->ExclusiveStore.Size` is correct. According to ARM documentation, a mismatched size between load and store exclusive is predictable undefined. `B2.12.5 Load-Exclusive and Store-Exclusive instruction usage restrictions` for the full documentation there.